pkg/innerring/processors/governance/list.go

```diff
@@ -62,3 +65,29 @@ func newAlphabetList(sidechain, mainnet keys.PublicKeys) (keys.PublicKeys, error
 
 	return result, nil
 }
+
+// updateInnerRing function removes `before` keys from `innerRing` and adds
+// `after` keys in the list. If length of `before` and `after` is not the same
+// then function returns errNotEqualLen
+func updateInnerRing(innerRing, before, after keys.PublicKeys) (keys.PublicKeys, error) {
+	lnBefore := len(before)
+	if lnBefore != len(after) {
+		return nil, errNotEqualLen
+	}
+
+	result := make(keys.PublicKeys, 0, len(innerRing))
+
+	// O(n^2) for 7 nodes is not THAT bad.
+loop:
+	for i := range innerRing {
+		for j := range before {
+			if innerRing[i].Equal(before[j]) {
+				result = append(result, after[j])
```

Comment:
So we replace elements from `before` with the elements from `after` with the exact same indices, is this really what it takes?

Reply:
Yeah. Indices are not important though, we just need to replace all `before` elements with `after` elements in any order. There is a guarantee that original list contains all `before` elements (if it isn't then algorithm is broken on higher levels).